cobigen/cobigen-core-api/src/main/java/com/devonfw/cobigen/api/util/to/MavenSearchResponse.java

```diff
@@ -0,0 +1,236 @@
+package com.devonfw.cobigen.api.util.to;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.devonfw.cobigen.api.constants.MavenSearchRepositoryConstants;
+import com.devonfw.cobigen.api.constants.MavenSearchRepositoryType;
+import com.devonfw.cobigen.api.exception.RESTSearchResponseException;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * Json model for maven Search REST API response
+ *
+ */
+@JsonIgnoreProperties(value = { "responseHeader", "spellcheck" })
+public class MavenSearchResponse implements SearchResponse {
+
+  /** Logger instance. */
+  @JsonIgnore
+  private static final Logger LOG = LoggerFactory.getLogger(MavenSearchResponse.class);
+
+  @JsonProperty("response")
+  private MavenSearchResponseResponse response;
+
+  /**
+   * @return response
+   */
+  @JsonIgnore
+  public MavenSearchResponseResponse getResponse() {
+
+    return this.response;
+  }
+
+  @Override
+  @JsonIgnore
+  public String getJsonResponse(String repositoryUrl, String groupId) throws RESTSearchResponseException {
+
+    return getJsonResponse(repositoryUrl, groupId, null);
+  }
+
+  @Override
+  @JsonIgnore
+  public String getJsonResponse(String repositoryUrl, String groupId, String authToken)
+      throws RESTSearchResponseException {
+
+    String targetLink = repositoryUrl + "/" + MavenSearchRepositoryConstants.MAVEN_TARGET_LINK + "?q=g:" + groupId
+        + "&wt=json";
+    LOG.info("Starting {} search REST API request with URL: {}.", getRepositoryType(), targetLink);
+
+    int limitRows = MavenSearchRepositoryConstants.MAVEN_MAX_RESPONSE_ROWS;
+    if (limitRows > 0) {
```

Comment:
Isn't that always true, no matter what? I don't understand the point of that if statement.

Reply:
This needs to be discussed as I'm not sure how to handle the optional row limitation of the maven search properly.
I could add a new limitRows param to the method signature or ignore the rows param completely.

Reply:
I've removed it now as this limit rows param is not needed for small search requests.